lib/knapsack_pro/test_case_detectors/rspec_test_example_detector.rb

```diff
@@ -3,87 +3,83 @@
 module KnapsackPro
   module TestCaseDetectors
     class RSpecTestExampleDetector
-      def generate_json_report(rspec_args)
-        raise "The internal KNAPSACK_PRO_RSPEC_OPTIONS environment variable is unset. Ensure it is not overridden accidentally. Otherwise, please report this as a bug: #{KnapsackPro::Urls::SUPPORT}" if rspec_args.nil?
-
-        require 'rspec/core'
-
-        cli_format =
-          if Gem::Version.new(::RSpec::Core::Version::STRING) < Gem::Version.new('3.6.0')
-            require_relative '../formatters/rspec_json_formatter'
-            ['--format', KnapsackPro::Formatters::RSpecJsonFormatter.to_s]
-          else
-            ['--format', 'json']
-          end
-
-        ensure_report_dir_exists
-        remove_old_json_report
-
-        test_file_entities = slow_test_files
-
-        if test_file_entities.empty?
-          no_examples_json = { examples: [] }.to_json
-          File.write(report_path, no_examples_json)
-          return
-        end
+      def calculate(rspec_args)
+        KnapsackPro::Config::TempFiles.ensure_temp_directory_exists!
+        FileUtils.mkdir_p(File.dirname(report_path))
+        File.delete(report_path) if File.exist?(report_path)
+        return File.write(report_path, { examples: [] }.to_json) if slow_test_files.empty?
 
+        KnapsackPro.logger.info("Calculating split by test examples. Analyzing #{slow_test_files.size} slow test files.")
         args = (rspec_args || '').split
         cli_args_without_formatters = KnapsackPro::Adapters::RSpecAdapter.remove_formatters(args)
-
-        # Apply a --format option which overrides formatters from the RSpec custom option files like `.rspec`.
```

Comment:
recommendation


```
Apply a --format option which overrides formatters from the RSpec custom option files like `.rspec`.
```

We could keep this comment, as this was important knowledge about RSpec behaviour. Or move the comment to the `cli_format` method.